readium/shared/src/main/java/org/readium/r2/shared/util/Instant.kt

```diff
@@ -0,0 +1,111 @@
+package org.readium.r2.shared.util
+
+import android.os.Parcel
+import android.os.Parcelable
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.format
+import kotlinx.datetime.format.DateTimeComponents
+import kotlinx.datetime.toInstant
+import kotlinx.parcelize.Parceler
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.TypeParceler
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import org.readium.r2.shared.InternalReadiumApi
+import org.readium.r2.shared.extensions.tryOrNull
+
+/**
+ * A moment in time.
+ */
+@Parcelize
+@TypeParceler<kotlinx.datetime.Instant, InstantParceler>()
+@Serializable(with = InstantSerializer::class)
+public class Instant private constructor(
```

Comment:
I used a dedicated type instead of using straight `kotlinx.datetime.Instant` to avoid requiring to leak it in our API with `api(libs.kotlinx.datetime)`.